processing/src/main/java/io/druid/segment/DimensionHandlerUtils.java

```diff
@@ -233,7 +233,8 @@ public static Long convertObjectToLong(Object valObj)
     } else if (valObj instanceof Number) {
       return ((Number) valObj).longValue();
     } else if (valObj instanceof String) {
-      return DimensionHandlerUtils.getExactLongFromDecimalString((String) valObj);
+      Long parsedVal = DimensionHandlerUtils.getExactLongFromDecimalString((String) valObj);
```

Comment:
Why null checks are needed in checkUnsortedEncodedKeyComponentsEqual() and checkUnsortedEncodedKeyComponentHashCode(), since you added it here?